### DISCUSSION TITLE
When will NotifyIcon in Notification Area be supported in WinUI3?

### DISCUSSION THREAD

**Discussion by rahomi** [microsoft/WindowsAppSDK#3394] [Q&A]
Currently, I did not find any support for NotifyIcon in WinUI3. However it is supported in WinForm/WPF. Is there any plan from Microsoft to support this in near future?

**Comment by castorix** [comment]
WInforms/WPF use [Shell_NotifyIcon](https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shell_notifyiconw), which is the normal method and MS probably thinks it is not a priority, beause it is a few lines of code to add it in WinUI 3 (in C#, you can just copy the declarations and parts of code with _Shell_NotifyIcon_ from WPF source code (_NativeMethods.cs_ and _NotifyIcon.cs_) then add _SetWindowSubclass_ for [uCallbackMessage](https://learn.microsoft.com/en-us/windows/win32/api/shellapi/ns-shellapi-notifyicondataw#nif_message-0x00000001))

**Comment by castorix** [comment]
There are some samples on GitHub; I added another one by using an Owner-Drawn Context Menu, to add a bit of code, otherwise it is just a function call to add icon or notification... : [WinUI3_NotifyIcon](https://github.com/castorix/WinUI3_NotifyIcon/tree/master)
